test/integration/targets/known_hosts/tasks/main.yml

```
@@ -61,6 +61,38 @@
     - 'known_hosts.stdout_lines[5].startswith("# example.net")'
     - 'known_hosts.stdout_lines[-1].strip() == example_org_rsa_key.strip()'
 
+# https://github.com/ansible/ansible/issues/84952
+# test addition when ~/.ssh/ directory missing and path=default
+- name: check if ~/.ssh exists
+  stat:
+    path: ~/.ssh
+  register: default_ssh_dir
+
+- name: move ~/.ssh/ directory away
+  command: mv ~/.ssh/ ~/.ssh_bak/
+  when: default_ssh_dir.stat.exists
+  register: backup_default_ssh_dir
+
+- name: add a new host with default path
+  known_hosts:
+    name: example.org
+    key: "{{ example_org_rsa_key }}"
+    state: present
+  register: result
+  ignore_errors: True
+
+- name: remove ~/.ssh/ directory created by known_hosts module
+  command: rm -r ~/.ssh
```

To ensure the next cleanup task runs too,
```suggestion
  command: rm -r ~/.ssh
  ignore_errors: True
```